demos/portal-embed-demo/public/js/ttt/audio-controller.js

```diff
@@ -0,0 +1,369 @@
+            this.axis.style.left = `${left}px`;
+        }
+    }
+
+    /**
+     * Adding event listeners for the buttons
+     */
+    _hookEvents() {
+        this.playButton.addEventListener('click', evt => {
+            if(this._isPlaying()){
+                this.pause();
+                return;
+            }
+            this.play();
+        });
+        this.nextButton.addEventListener('click', evt => this.next());
+        this.prevButton.addEventListener('click', evt => this.prev());
+        this.audio.addEventListener("ended", evt => {
+            this.pause();
+            this._createNewAudio(this.playlist[this.playlistIndex]);
+       });
+    }
+
+    /**
+     * Creating a new audio element
+     * @param {String} id - audio ID
+     */
+    _createNewAudio(id) {
+
+        // Simply replace the exisiting audio element with a new one
+        if(this.root.querySelector('audio')){
+            this.root.removeChild(this.root.querySelector('audio'));
+        }
+        const audio = document.createElement('audio');
+        const source = document.createElement('source');
+
+        // We could build streaming API for this in the future
+        source.src = `/mp3/${id}.mp3`;
+        audio.appendChild(source);
+        this.root.appendChild(audio);
+        this.audio = audio;
+
+        // Resetting the progress bar
+        this.durationSec = audioList[id].durationSec;
+        this.audioTitle.innerHTML
+            = `${audioList[id].title} - Totally Tooling Tips`;
+        this.axis.style.transition = '';
+        this.axis.style.left = '0px';
+    }
+
+    /**
+     * Update the prev/next button status
+     */
+    _updateController() {
+        if(this._hasPrev()) {
+            this.prevButton.style.opacity = 1.0;
+        } else {
+            this.prevButton.style.opacity = 0.3;
+        }
+        if(this._hasNext()){
+            this.nextButton.style.opacity = 1.0;
+        } else {
+            this.nextButton.style.opacity = 0.3;
+        }
+    }
+
+    /**
+     * Checks if the controller has a previous audio to play
+     * @returns {boolean}
+     */
+    _hasPrev() {
+        return this.playlistIndex !== 0;
+    }
+
+    /**
+     * Checks if the controller has a next audio to play
+     * @returns {boolean}
+     */
+    _hasNext() {
+        return this.playlist.length > 1 
+            && !(this.playlistIndex + 1 >= this.playlist.length);
+    }
+
+    /**
+     * Calculates the current left position of the animated axis
+     * @returns {Number}
+     */
+    _getAxisCurrentLeft() {
+        const timelineWidth = this.timeline.offsetWidth;
+        const playerWidth = this.audioTitle.offsetWidth;
+        const offset = (playerWidth - timelineWidth)/2;
+        const rectLeft = this.axis.getBoundingClientRect().left;
+        return rectLeft - offset;
+    }
+
+    /**
+     * Checks if the controller is currently playing an audio
+     * @returns {boolean}
+     */
+    _isPlaying() {
```

Comment:
Yes. Initially I was managing the state with a custom attribute set to the play button like `this.button.setAttribute("isPlaying", true/false)` and wanted to have that operation/checking in a separate function. But since I ended up using the class member, I'll just remove this function.